aquacomputer_d5next.c

```diff
@@ -869,23 +940,43 @@ static int aqc_read(struct device *dev, enum hwmon_sensor_types type, u32 attr,
 			*val = *val + 1;
 			break;
 		case hwmon_pwm_input:
-			if (priv->kind == aquaero) {
+			switch (priv->kind) {
+			case aquaero:
 				ret =
 				    aqc_get_ctrl_val(priv,
 						     AQUAERO_CTRL_PRESET_START +
 						     channel * AQUAERO_CTRL_PRESET_SIZE, val, 16);
 				if (ret < 0)
 					return ret;
-			} else {
+				*val = aqc_percent_to_pwm(*val);
+				break;
+			case aquastreamxt:
+				if (channel == 0) {
+					ret =
+					    aqc_get_ctrl_val(priv, priv->fan_ctrl_offsets[channel],
+							     val, 32);
+					if (ret < 0)
+						return ret;
+					*val = aquastream_convert_pump_rpm(*val);
+					*val = (*val - 3000) * 255 / 3000;
```

Comment:
Note that this is scaling the value? And `3000` should be in a macro, like `AQUASTREAMXT_PUMP_MIN_RPM`.